### PR TITLE
Add example for performing a prefetch

### PR DESCRIPTION
**Pull request by andon**

### CODE REVIEW

File: packages/core/src/zip/index.js [hunk 1]
```diff
@@ -1,5 +1,15 @@
 'use strict'
 
+import { visit, onPre } from '../vendor/zippa'
+
 export * from '../vendor/zippa'
 
 export { default as elementZipper } from './elementZipper'
+
+const reducingVisitor = fn =>
```
Comment (ognen) [inline packages/core/src/zip/index.js]: name it just `reduce`, `preReduce`, add `postReduce`

File: packages/core/src/zip/index.js [hunk 2]
```diff
@@ -1,5 +1,15 @@
...
+  onPre((item, state) => ({ state: fn(state, item) }))
+
+export const when = (pred, fn) => (acc, i) => (pred(i) ? fn(acc, i) : acc)
```
Comment (ognen) [inline packages/core/src/zip/index.js]: Move it to `data`. It has nothing to do with zippers.

File: packages/core/src/__tests__/prefetch.js [hunk 3]
```diff
@@ -0,0 +1,149 @@
+'use strict'
```
Comment (ognen) [inline packages/core/src/__tests__/prefetch.js]: I don't get why this test should go into the the test suite?

Comment (ognen) [reply packages/core/src/__tests__/prefetch.js]: If this is an example, please move it in examples or documentation.

Comment (andon) [reply packages/core/src/__tests__/prefetch.js]: Yes, example, didn't plan to keep it in, just to challenge / discuss the issue.

Comment (andon) [reply packages/core/src/__tests__/prefetch.js]: @ognen do you want to move it to example tests, or just a create a markdown doc page in `docs`?

**Comment by andon** [comment]
Not needed any more. `zip.reduce` implemented now with https://github.com/netceteragroup/girders-elements/pull/88

**Pull request closed without merging**